gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java

```diff
@@ -15,258 +15,68 @@
  */
 package io.gravitee.am.certificate.javakeystore.provider;
 
-import com.nimbusds.jose.jwk.JWKSet;
-import com.nimbusds.jose.jwk.KeyUse;
-import io.gravitee.am.certificate.api.CertificateFormat;
-import io.gravitee.am.certificate.api.CertificateKey;
+import io.gravitee.am.certificate.api.AbstractCertificateProvider;
 import io.gravitee.am.certificate.api.CertificateMetadata;
-import io.gravitee.am.certificate.api.CertificateProvider;
-import io.gravitee.am.certificate.api.DefaultKey;
-import io.gravitee.am.certificate.api.RSAKeyUtils;
-import io.gravitee.am.certificate.api.X509CertUtils;
 import io.gravitee.am.certificate.javakeystore.JavaKeyStoreConfiguration;
-import io.gravitee.am.common.jwt.SignatureAlgorithm;
-import io.gravitee.am.model.jose.JWK;
-import io.gravitee.am.model.jose.RSAKey;
-import io.reactivex.Flowable;
-import io.reactivex.Single;
 import org.springframework.beans.factory.InitializingBean;
 import org.springframework.beans.factory.annotation.Autowired;
 
-import java.io.ByteArrayInputStream;
-import java.io.InputStream;
-import java.security.Key;
-import java.security.KeyPair;
 import java.security.KeyStore;
-import java.security.PrivateKey;
-import java.security.cert.Certificate;
-import java.security.cert.X509Certificate;
-import java.security.interfaces.RSAPrivateKey;
-import java.security.interfaces.RSAPublicKey;
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.List;
-import java.util.Objects;
-import java.util.Optional;
+import java.security.KeyStoreException;
 import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 /**
  * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
  * @author GraviteeSource Team
  */
-public class JavaKeyStoreProvider implements CertificateProvider, InitializingBean {
-
-    private Date expirationDate;
-    private KeyPair keyPair;
-    private Certificate cert;
-    private JWKSet jwkSet;
-    private Set<JWK> keys;
-    private SignatureAlgorithm signature = SignatureAlgorithm.RS256;
-    private io.gravitee.am.certificate.api.Key certificateKey;
-    private List<CertificateKey> certificateKeys;
-
+public class JavaKeyStoreProvider extends AbstractCertificateProvider implements InitializingBean {
     @Autowired
     private JavaKeyStoreConfiguration configuration;
-
     @Autowired
     private CertificateMetadata certificateMetadata;
```

Comment:
Is it possible to considere this attribute to be part of the Abstract class as it is also used into the PKCS12Provider ?

Reply:
I can do that, so long as adding the Spring dependency to that module isn't a problem?

Reply:
As Spring is used on REST API & GW, I don't think it is a problem IMHO.